python/pyarrow/parquet.py

```diff
@@ -985,7 +985,7 @@ def write_to_dataset(table, root_path, partition_cols=None,
     else:
         fs = _ensure_filesystem(filesystem)
 
-    if not fs.exists(root_path):
+    if not fs.exists(root_path) and fs.isfilestore():
```

Comment:
Would be better to switch the checks around

```
 if fs.isfilestore() and not fs.exists(root_path) :
```

That way we don't do an unnecessary `exists()` check if it's not a file store.`

Reply:
Yes, great point! Updated.